addons/better-terrain/BetterTerrain.gd

```diff
@@ -583,6 +583,22 @@ func get_tile_terrain_type(td: TileData) -> int:
 	return td_meta.type
 
 
+## Returns an Array of all [TileData] tiles included in the specified
+## terrain [code]type[/code] for the [TileSet] [code]ts[/code]
+func get_tiles_in_terrain(ts: TileSet, type: int) -> Array[TileData]:
+	var result:Array[TileData] = []
+	
+	var cache := _get_cache(ts)
+	var tiles = cache[type]
+	if tiles and tiles.size() > 0:
```

Comment:
I would prefer
```
if !tiles:
    return result
```
to early return, and you don't need to check the size because no iteration will occur on an empty array